### PR TITLE
Tsc

### PR DESCRIPTION
**Pull request by kberg**
Key changes:
1. styles.css is moved to build, so .gitignore can ignore it.
2. the two codegenned JSON files are moved to src/genfiles. the clean command and .gitignore changed appropriately.
3. Moved tsconfig.json into src, and tsconfig-test.json into tests
4. Converted both of those files so they would use project references.

One upshot of all of this is once you clone the repo you can run
```
$ npm run install
$ npm run test
```

Edited to add: this also restores test files to full participation in the tserver index.

### CODE REVIEW

**Comment by Lynesth** [comment]
I missed a PR apparently, but why was the linter moved to the last spot ?
Can it be back to the front ?

**Comment by Lynesth** [comment]
I'm not sure why you're moving settings and translations, but I think you also need to if you're doing it I think you also need to fix the routes for translations in server.ts.

File: src/tsconfig.json [hunk 1]
```diff
@@ -0,0 +1,12 @@
+{
+    "compilerOptions": {
+        "outDir": "../build/src",
+        "module": "commonjs",
```
Comment (Lynesth) [inline src/tsconfig.json:4]: Is it needed since it's already declared in the base tsconfig ?

**Comment by kberg** [comment]
> I missed a PR apparently, but why was the linter moved to the last spot ?
> Can it be back to the front ?

That was done slightly recently, because it's particularly slow -- I'd rather get compiler errors first.

**Comment by kberg** [comment]
> > I missed a PR apparently, but why was the linter moved to the last spot ?
> > Can it be back to the front ?
> 
> That was done slightly recently, because it's particularly slow -- I'd rather get compiler errors first.

I'm not opposed to it, though.

**Comment by kberg** [comment]
> I'm not sure why you're moving settings and translations, but if you're doing it I think you also need to fix the routes for translations in server.ts.

Yipe! This was in an earlier version I swear. I'll fix that immediately.

It's to put all the _source_ in _src_. And because the `src/tsconfig.json` must have all code in its root directory.

File: src/tsconfig.json (hunk 1 above)
Comment (kberg) [reply src/tsconfig.json:4]: Thanks, fixed.

File: make_static_json.js [hunk 2]
```diff
@@ -56,12 +58,15 @@ function getWaitingForTimeout() {
     return 5000;
 }
 
-fs.writeFileSync("assets/settings.json", JSON.stringify({
+if (!fs.existsSync('src/genfiles')) {
+    fs.mkdirSync('src/genfiles');
```
Comment (bafolts) [inline make_static_json.js:62]: Using these is helpful for ensuring this builds on every OS :+1:

File: src/components/App.ts [hunk 3]
```diff
@@ -10,7 +10,7 @@ import {DebugUI} from './DebugUI';
 import {GameHomeModel} from '../models/GameHomeModel';
 import {HelpIconology} from './HelpIconology';
 
-import * as raw_settings from '../../assets/settings.json';
+import * as raw_settings from '../genfiles/settings.json';
```
Comment (bafolts) [inline src/components/App.ts:13]: This is a great name change. Makes it clearer that we generate these files.

**Review by bafolts — APPROVED** [review]

**Comment by bafolts** [comment]
I often run `npm run build` while I still have linting errors. Having the lint step last works for my workflow.

**Comment by Lynesth** [comment]
Could we add --cache to the eslint command then ?

**Comment by kberg** [comment]
> Could we add --cache to the eslint command then ?

Feel free! It sounds like we should have done that in the first place.

**Comment by bafolts** [comment]
I may add the `--cache` for `eslint`. I work on underpowered linux machines and that could be a game changer.

**Pull request merged**